packages/eas-cli/src/submissions/utils/builds.ts

```diff
@@ -1,30 +1,35 @@
-import { AppPlatform, BuildStatus } from '../../graphql/generated';
+import { AppPlatform, BuildArtifacts, BuildFragment, BuildStatus } from '../../graphql/generated';
 import { BuildQuery } from '../../graphql/queries/BuildQuery';
 
-export async function getBuildArtifactUrlByIdAsync(
+// `BuildFragment` with non-null `artifacts.buildUrl`
+type BuildFragmentWithArtifact = Omit<BuildFragment, 'artifacts'> & {
+  artifacts: Omit<BuildArtifacts, 'buildUrl'> & BuildArtifacts & { buildUrl: string };
+};
```

Comment:
```suggestion
type BuildFragmentWithArtifact = BuildFragment & {
  artifacts: BuildArtifacts & { buildUrl: string };
};
```
not tested, but it should work